erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py

```diff
@@ -298,12 +301,40 @@ def on_submit(self):
 			from erpnext.stock.doctype.serial_no.serial_no import update_serial_nos_after_submit
 			update_serial_nos_after_submit(self, "items")
 
+		if self.update_purchase_receipt == 1:
+			self.update_sl_valuation()
+
 		# this sequence because outstanding may get -negative
 		self.make_gl_entries()
 
 		self.update_project()
 		self.update_fixed_asset()
 
+	def update_sl_valuation(self):
+		for item in self.get("items"):
+			pr = item.purchase_receipt
+			pr_doc = frappe.get_doc("Purchase Receipt", pr)
+		
+			for pr_item in pr_doc.get("items"):
+				if item.item_code == pr_item.item_code:
+					pr_item.valuation_rate = item.rate
+
+			# save will update item valuation rate in PR
+			pr_doc.save()
```

Comment:
Instead of saving, use `db_update` method to avoid unwanted validations and mandatory checks. Check for reference, https://github.com/frappe/erpnext/blob/master/erpnext/stock/doctype/landed_cost_voucher/landed_cost_voucher.py#L114